### PR TITLE
Support channel updates

### PR DESCRIPTION
**Pull request by michaelbausor**
- Add support for channel and forceNewChannel args
- Support updated grpcCredentialsHelper
- Set stubs and grpcCredentialsHelper to protected to allow access from partial veneers

cc @dwsupplee @jdpedrie

### CODE REVIEW

**Comment by michaelbausor** [comment]
@garrettjonesgoogle @shinfan @neozwu Hi all, gentle ping on this after fixit week

File: src/main/resources/com/google/api/codegen/php/client_impl.snip [hunk 1]
```diff
@@ -346,13 +345,23 @@
...
      @end
      *     @@type mixed $port The port on which to connect to the remote host. Default {@xapiClass.servicePort}.
+     *     @@type \Grpc\Channel $channel
+     *           A `Channel` object to be used by gRPC. If not specified, a channel will be constructed.
      *     @@type \Grpc\ChannelCredentials $sslCreds
-     *           A `ChannelCredentials` for use with an SSL-enabled channel.
+     *           A `ChannelCredentials` object for use with an SSL-enabled channel.
      *           Default: a credentials object returned from
      *           \Grpc\ChannelCredentials::createSsl()
+     *           NOTE: if the $channel optional argument is specified, then this option is unused.
```
Comment (shinfan) [inline src/main/resources/com/google/api/codegen/php/client_impl.snip]: nit: s/this option/this argument/

File: src/test/java/com/google/api/codegen/testdata/php_client_impl_library.baseline [hunk 2]
```diff
@@ -455,12 +454,22 @@ class LibraryServiceGapicClient
...
+     *           A `ChannelCredentials` object for use with an SSL-enabled channel.
      *           Default: a credentials object returned from
      *           \Grpc\ChannelCredentials::createSsl()
+     *           NOTE: if the $channel optional argument is specified, then this option is unused.
+     *     @type bool $forceNewChannel
+     *           If true, this forces gRPC to create a new channel instead of using a persistent channel.
+     *           Defaults to false.
+     *           NOTE: if the $channel optional argument is specified, then this option is unused.
+     *     @type \Google\Auth\CredentialsLoader $credentialsLoader
+     *                              A CredentialsLoader object created using the
```
Comment (shinfan) [inline src/test/java/com/google/api/codegen/testdata/php_client_impl_library.baseline]: Is the indentation incorrect here?

File: src/main/resources/com/google/api/codegen/php/client_impl.snip (hunk 1 above)
Comment (michaelbausor) [inline src/main/resources/com/google/api/codegen/php/client_impl.snip]: Done

File: src/test/java/com/google/api/codegen/testdata/php_client_impl_library.baseline (hunk 2 above)
Comment (michaelbausor) [inline src/test/java/com/google/api/codegen/testdata/php_client_impl_library.baseline]: No, good point, fixed. (This is fixed outside toolkit by the php formatting tools, but we should still get it correct here).

**Comment by michaelbausor** [comment]
PTAL

**Pull request merged**